app/pages/lab/workflow-viewer/task.cjsx

```diff
@@ -0,0 +1,313 @@
+React = require 'react'
+Resizable = require 'react-component-resizable'
+{TaskName, RequireBox, HelpButton} = require './task-helpers'
+AnswerList = require './answers'
+Sty = require './jsplumb-style'
+
+# keep track of the max z-index
+max_z = 0
+
+# Firefox needs an offset patch for jsPlumb elements (not sure way)
+isFirefox = typeof InstallTrigger != 'undefined'
+
+# timer to detect when resize stops
+timer = null
+
+# Make a task node
+# Takes `plumbId`, `workflow`, `taskKey`, `pos`, and `taskNumber` as a props
+Task = React.createClass
+  displayName: 'Task'
+
+  getInitialState: ->
+    zIndex: @props.taskNumber + 2
+    task: @props.wf[@props.taskKey]
+    endpoints: []
+    zIndex: @props.taskNumber + 2
+    uuid: 0
+    uuids: []
+    me: null
+    width:
+      min: 0
+      max: 0
+      val: 0
```

Comment:
Couple of things here. First, there are few are far between instances where you should immediately move a prop into a state variable. Doing that means you are now tracking app state in two spots (this component and the parent component), which can lead to state synchronization bugs. You should be able to replace every instance of `@state.zIndex` with `@props.taskNumber + 2` or if you want to be more explicit about it, using a function like so:

```
computeZIndex = (taskNumber) ->
  taskNumber + 2
```

Same comments as above on variable naming.

Be careful about duplicate keys within the object you are returning. In this case, only one of the `zIndex` keys matters (doesn't really matter here, but might somewhere else).